Libraries/LibWeb/HTML/HTMLImageElement.cpp

```diff
@@ -170,22 +170,38 @@ bool HTMLImageElement::is_image_available() const
 
 Optional<CSSPixels> HTMLImageElement::intrinsic_width() const
 {
-    if (auto image_data = m_current_request->image_data())
+    if (auto image_data = m_current_request->image_data()) {
+        if (auto bitmap = current_image_bitmap()) {
+            return bitmap->width_considering_exif();
+        }
+
         return image_data->intrinsic_width();
+    }
+
     return {};
```

Comment:
changing HTMLImageElement to look into bitmap directly instead of going trough DecodedImageData seems wrong because there are also HTMLObjectElement and HTMLInputElement which are not aware of this change. can we instead update DecodedImageData to get size with orientation applied?